src/player.cpp

```diff
@@ -1159,7 +1159,8 @@ void player::update_bodytemp()
     }
 
     const int lying_warmth = use_floor_warmth ? floor_warmth( pos() ) : 0;
-    const int water_temperature = 100 * temp_to_celsius( g->get_cur_weather_gen().get_water_temperature() );
+    const int water_temperature =
+        100 * ( int )temp_to_celsius( g->get_cur_weather_gen().get_water_temperature() );
```

Comment:
This changes the behaviour, the variable will now always hold multiples of 100, but never anything between 101 and 199 (for example).

Reply:
Ah good catch.